File: channel_gain.m
function g_d = channel_gain(Freq_THz,Comm_dist,Skin_thickness,Tissue_thickness,Vessel_thickness)
    g_d=0.0;
    coder.updateBuildInfo('addSourceFiles','terahertz.c');
    coder.cinclude('terahertz.h'); 
    g_d=coder.ceval('signalAttenuation',Freq_THz,Comm_dist,Skin_thickness,Tissue_thickness,Vessel_thickness);

end

File: codegen/mex/channel_gain/channel_gain.c
/*
 * Academic License - for use in teaching, academic research, and meeting
 * course requirements at degree granting institutions only.  Not for
 * government, commercial, or other organizational use.
 *
 * channel_gain.c
 *
 * Code generation for function 'channel_gain'
 *
 */

/* Include files */
#include "channel_gain.h"
#include "rt_nonfinite.h"
#include "terahertz.h"

/* Function Definitions */
real_T channel_gain(const emlrtStack *sp, real_T Freq_THz, real_T Comm_dist,
                    real_T Skin_thickness, real_T Tissue_thickness,
                    real_T Vessel_thickness)
{
  (void)sp;
  return signalAttenuation(Freq_THz, Comm_dist, Skin_thickness,
                           Tissue_thickness, Vessel_thickness);
}

/* End of code generation (channel_gain.c) */
